Engagement refactoring w awarding

eri-trabiccolo opened the pull request:
continue working on the engagement awarding

eri-trabiccolo committed:
Fix select and hidden fields not rendering required attribute

eri-trabiccolo committed:
better handling eangagement awarding on creation

eri-trabiccolo committed:
Retrieve earned dates using the LLMS_User_Achievement(Certificate) model

thomasplevy commented:
Some notes:

1.  When viewing an earned cert as an admin I'd expect to see the "Edit Certificate" link/button in the WP Admin bar (and I don't)
1. When editing an earned cert / achievement (on the admin panel) the "Engagements" menu in the sidebar collapses. Ideally this would stay expanded
1. When awarding a cert/achiemvent (creating a new my_* manually on the admin panel) the post type language switches from "Award Certificate" to "Add new certificate". I think it should say "Award New Certificate" and "Edit Awarded Certificate"
1. When editing an awarded cert / engagement we should add the student's name/email and a link to their reporting screen in a metabox somewhere

eri-trabiccolo commented:
I agree on everything -
1,2,4 were on my list too.
3 added. 
❤️

eri-trabiccolo commented:
I'm actually also planning to rework the submit box and add our own JavaScript because the wp JS forces the wording (e.g. Publish, while I want Award).

eri-trabiccolo committed:
Add award engagements submit box

eri-trabiccolo committed:
Keep the Engagements admin menu expanded when editing an awarded engagement
plus show "Edit Certificate" link/button in the WP Admin bar when wiewing
an earned certificate.

eri-trabiccolo commented:
> 1. When viewing an earned cert as an admin I'd expect to see the "Edit Certificate" link/button in the WP Admin bar (and I don't)
> 2. When editing an earned cert / achievement (on the admin panel) the "Engagements" menu in the sidebar collapses. Ideally this would stay expanded

Fixed
> 3. When awarding a cert/achiemvent (creating a new my_* manually on the admin panel) the post type language switches from "Award Certificate" to "Add new certificate". I think it should say "Award New Certificate" and "Edit Awarded Certificate"

Fixed

> 4. When editing an awarded cert / engagement we should add the student's name/email and a link to their reporting screen in a metabox somewhere

Fixed

Added it in the same metabox, not exactly happy of the look and feel to be honest, what do you think? Do you think it deserves its only metabox? Also consider that when your work will be done, there will be no other fields in the metabox, as both title (and content, for the achievement) and background will be the WordPress default one.

eri-trabiccolo committed:
Normalize awarded engagements post types language

eri-trabiccolo committed:
Remove earned engagements post types double registration

eri-trabiccolo committed:
Trait usage doesn't need emphasis

eri-trabiccolo committed:
When eediting an awarded cert / engagement add the student's name/email and a link to their reporting screen

eri-trabiccolo committed:
move engagement awarding actions in the submit metabox class

eri-trabiccolo committed:
Handle llms_my_certificate and llms_my_certificate post types update messages

thomasplevy commented on includes/class.llms.post-types.php:
```diff
@@ -1456,34 +1444,62 @@ public static function register_taxonomies() {
 	 * @return void
 	 */
-	public static function add_earned_egnagements_submenu_links() {
+	public static function modify_earned_engagements_submenu_links() {
```
I think this is better suited in this file: https://github.com/gocodebox/lifterlms/blob/trunk/includes/admin/class.llms.admin.menus.php

eri-trabiccolo committed:
Move admin menu modifications in
includes/admin/class.llms.admin.menus.php

eri-trabiccolo committed:
llms_my_achievement supports editor

eri-trabiccolo committed:
Move earned engagement trait to award engagement mb

eri-trabiccolo commented:
wanna merge this in your branch?
